apps/frontend/src/components/CurrentCharity.tsx

```diff
@@ -0,0 +1,127 @@
+import { useEffect, useState } from "react";
+import { Typography, Grid, Box, Avatar, Paper } from "@mui/material";
+import { colors } from "../styles/colors";
+import { Charity } from "../pages/Charities";
+
+interface CurrentCharityProps {
+  charity: Charity;
+}
+
+const CurrentCharity = ({ charity }: CurrentCharityProps) => {
+  const [timeRemaining, setTimeRemaining] = useState<string>("");
+
+  useEffect(() => {
+    const calculateTimeRemaining = () => {
+      const endTime = new Date(charity.endDate).getTime();
+      const now = new Date().getTime();
+      const distance = endTime - now;
+
+      if (distance < 0) {
+        return "This charity event has ended.";
+      } else {
+        const days = Math.floor(distance / (1000 * 60 * 60 * 24));
+        const hours = Math.floor(
+          (distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60),
+        );
+        const minutes = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60));
+        const seconds = Math.floor((distance % (1000 * 60)) / 1000);
+        return `${days}d ${hours}h ${minutes}m ${seconds}s`;
+      }
+    };
```

Comment:
This is a really cool functionality! However, it does take a bit to render even on my beefy desktop computer, and it re-renders every time the toggle is pressed, it doesn't personally bother me, but if you're feeling keen to work on it, useCallback might be of some help

![image](https://github.com/user-attachments/assets/81a00b5d-1ec6-48df-8159-d2662631ea70)
![image](https://github.com/user-attachments/assets/abc933ce-ca39-458b-98bd-d1e75b417545)